lib/jsdom/living/events/EventTarget-impl.js

```diff
@@ -234,6 +241,48 @@ function invokeEventListeners(listeners, target, eventImpl) {
 
 const wrappedListener = Symbol("inline event listener wrapper");
 
+/**
+ * Normalize the event listeners options argument in order to get always a valid options object
+ * @param   {Object} options         - user defined options
+ * @param   {Array} defaultBoolKeys  - boolean properties that should belong to the options object
+ * @param   {Array} keysBlackList    - options not allowed (optional)
+ * @returns {Object} object containing at least the  "defaultBoolKeys"
+ */
+function normalizeEventHandlerOptions(options, defaultBoolKeys, keysBlackList) {
+  // normalize the options argument in order to get always an object
+  if (options === null || options === undefined) {
+    options = {};
+  } else if (typeof options !== "object") {
+    options = { capture: options };
+  }
+
+  let defaultOptions;
+  let additionalOptions;
+  keysBlackList = keysBlackList || [];
+
+  try {
```

Comment:
This whole section is way overcomplicated. You should just write the following:

```js
const returnValue = {};

if (typeof options === "bool") {
  returnValue.capture = Boolean(options);
  return returnValue;
}

for (const key of defaultBoolKeys) {
  returnValue[key] = Boolean(options[key]);
}

return returnValue;
```

Don't catch any errors; don't bother with a blacklist; none of that. Users cannot pass other properties as options.